### PR TITLE
tomb

### PR DESCRIPTION
**Pull request by ysono**
- Un-ignore tombstones.
- Memtable->SSTable does not require re-reading

Big bug -- deletion weren't being persisted. We were deleting value out of memtable, so it would never go to sstable. I was conflating `Option<Value>` and tombstone. Now the latter is explicitly `Value::Tombstone`.

### CODE REVIEW

**Review by btc — APPROVED** [review]
i must admit i still have open questions about the serde layering but it's probably ok for now!

**Comment by btc** [comment]
good to go!

File: src/storage/serde.rs [hunk 1]
```diff
@@ -1,53 +1,57 @@
...
+}
+
+fn deserialize_val(bytes: Vec<u8>) -> Value {
+    Value::Bytes(bytes)
 }
 
 enum FileItem {
     EOF,
-    Item(usize, Option<Vec<u8>>),
+    Skip(usize),
```
Comment (btc) [inline src/storage/serde.rs:43]: perhaps a comment on what the different states represent. it's not clear to me what Skip and Empty denote.

**Comment by ysono** [comment]
> i must admit i still have open questions about the serde layering but it's probably ok for now!

I have a feeling it can be a lot better. Please feel free to refactor!

File: src/storage/serde.rs (hunk 1 above)
Comment (ysono) [reply src/storage/serde.rs:43]: Oh oops I missed your comment. Let me do this.

**Pull request merged**